python/ray/data/_internal/logical/operators/all_to_all_operator.py

```diff
@@ -0,0 +1,92 @@
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from ray.data._internal.block_list import BlockList
+from ray.data._internal.compute import BlockTransform
+from ray.data._internal.execution.interfaces import PhysicalOperator, RefBundle
+from ray.data._internal.execution.operators.all_to_all_operator import AllToAllOperator
+from ray.data._internal.logical.interfaces import LogicalOperator
+
+from ray.data._internal.stats import StatsDict
+
+
+class AbstractAllToAll(LogicalOperator):
+    """Abstract class for logical operators should be converted to physical
+    AllToAllOperator.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        input_op: LogicalOperator,
+        fn: Callable[[BlockList, bool, Callable, dict], Tuple[BlockList, dict]],
+        num_blocks: Optional[int] = None,
+        supports_block_udf: bool = False,
+        block_udf: Optional[BlockTransform] = None,
```

Comment:
Should we remove supports_block_udf / block_udf from the logical plan (afaik it's only used to represent fused stages, and we can instead represent it as Map->AllToAll)?

I think the planner can generate the code implementing the block udf fusion.

Reply:
@ericl - yeah removed.